modules/widgets/mailchimp/js/admin.js

```diff
@@ -0,0 +1,325 @@
+
+	var connectURL = '';
+	/**
+	 * Whether the site is connected to Mailchimp or no.
+	 */
+	var formWrapper = $( '.mailchimp-widget-jetpack-form-wrapper' );
+
+	/**
+	 * Check the connectivity with MailChimp.
+	 */
+	function apiCall() {
+		formWrapper.empty();
+		showLoading();
+		$.ajax( {
```

Comment:
We may want to add a nonce here and in other Ajax requests to avoid abuse.

Reply:
@jeherve Thanks for the suggestion, I agree, the nonces are added in the last commit.